### PR TITLE
add ability to build/publish specific docker images

### PR DESCRIPTION
**Pull request by yvdinesh**

### CODE REVIEW

**Comment by yvdinesh** [comment]
@nmiyake ready for review.

File: apps/distgo/cmd/docker/products.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		return false
+	}
+}
+
+func setToSlice(s map[string]struct{}) []string {
+	var result []string
+	for item := range s {
+		result = append(result, item)
+	}
+	return result
```
Comment (nmiyake) [inline apps/distgo/cmd/docker/products.go]: Should sort the result so that it's deterministic

File: apps/distgo/cmd/docker/products_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+				DockerImages: []params.DockerImage{
+					{
+						Repository: "test-repo",
+					},
+				},
+			},
+		},
+	}
+	_, _, err := productsToDistAndBuildImage([]string{"bar"}, project)
+	require.Error(t, err)
```
Comment (nmiyake) [inline apps/distgo/cmd/docker/products_test.go]: For error tests, it's typically better to have a regular expression that should match the error content as part of the test spec and assert that it matches after requiring the error (that way, you can verify that the failure is occurring for the expected reason)

File: apps/distgo/cmd/docker/products.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+func validateProducts(products []string, cfg params.Project) error {
+	var unknownProducts []string
+	for _, product := range products {
+		if _, ok := cfg.Products[product]; !ok {
+			unknownProducts = append(unknownProducts, product)
+		}
+	}
+	if len(unknownProducts) != 0 {
+		var filteredProducts []string
+		for product := range cfg.FilteredProducts() {
```
Comment (nmiyake) [inline apps/distgo/cmd/docker/products.go]: Does this return a sorted list/map? If not, should sort.

File: apps/distgo/cmd/docker/products.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+		}
+	}
+	return setToSlice(distProducts), setToSlice(imageProducts), nil
+}
+
+func validateProducts(products []string, cfg params.Project) error {
+	var unknownProducts []string
+	for _, product := range products {
+		if _, ok := cfg.Products[product]; !ok {
+			unknownProducts = append(unknownProducts, product)
```
Comment (nmiyake) [inline apps/distgo/cmd/docker/products.go]: Are products sorted?

**Comment by nmiyake** [comment]
Fixed issues and merged in https://github.com/palantir/godel/pull/221, so closing this PR.

**Pull request closed without merging**